pkg/swagger/generator.go

```diff
@@ -0,0 +1,140 @@
+package swagger
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+
+	internalapi "github.com/Azure/ARO-RP/pkg/api"
+	v20200430 "github.com/Azure/ARO-RP/pkg/api/v20200430"
+	"github.com/Azure/ARO-RP/pkg/api/v20210131preview"
+)
+
+type generator interface {
+	ExampleOpenShiftClusterPutParameter() interface{}
+	ExampleOpenShiftClusterPatchParameter() interface{}
+	ExampleOpenShiftClusterResponse() interface{}
+	ExampleOpenShiftClusterCredentialsResponse() interface{}
+	ExampleOpenShiftClusterAdminKubeconfigResponse() interface{}
+	ExampleOpenShiftClusterListResponse() interface{}
+	ExampleOperationListResponse() interface{}
+
+	Features() features
+}
+
+const apiv20200430Path = "github.com/Azure/ARO-RP/pkg/api/v20200430"
+const apiv20210131previewPath = "github.com/Azure/ARO-RP/pkg/api/v20210131preview"
+
+var _ generator = &apiv20210131preview{}
+var _ generator = &apiv20200430{}
+
+type features struct {
+	kubeconfig         bool
+	xmsEnum            bool
+	commonTypesVersion string
+	systemData         bool
+}
+
+type apiv20210131preview struct {
```

Comment:
I think this is pretty over engineered - you could probably simplify and get away with a static configuration like this:
```golang
type api struct {
	exampleOpenShiftClusterPutParameter func() interface{}
	exampleOpenShiftClusterPatchParameter func() interface{}
	exampleOpenShiftClusterResponse func() interface{}
	exampleOpenShiftClusterCredentialsResponse func() interface{}
	exampleOpenShiftClusterAdminKubeconfigResponse func() interface{}
	exampleOpenShiftClusterListResponse func() interface{}
	exampleOperationListResponse func() interface{}

	xmsEnum            []string
	commonTypesVersion string
	systemData         bool
}

var apis = map[string]*api {
	apiv20200430Path: {
		exampleOpenShiftClusterPutParameter: v20200430.ExampleOpenShiftClusterPutParameter,
		// ...
	},
	// ...
}

Reply:
Can we do this in go?
```
cannot use v20200430.ExampleOpenShiftClusterPutParameter (value of type func() *v20200430.OpenShiftCluster) as func() interface{} value in struct literal
```

was sitting on this for 2 hours trying to understand what you had in mind here....
```
foo func() interface{}
!=
func foo() *Custom{
```
https://play.golang.org/p/jUb0aqTLP5s

Reply:
Before I start refactoring for 4th time I would like to understand if Im missing something here for this to work (?)

Reply:
This caught my curiosity and found this interesting answer.
https://stackoverflow.com/a/54751503
I am still trying to digest all of it

Reply:
@mjudeikis I'm inclined to do the horrible thing and make all the Example*() functions return interface{}.  It's rubbish but it's a lot less boilerplate.